### DISCUSSION TITLE
Code Signing Certificates

### DISCUSSION THREAD

**Discussion by johnkattenhorn** [unoplatform/uno#13635] [General]
We are preparing to deliver 3 apps (1 x Win, 2 x Mobile Apps) we've built for a client using Uno, and they are currently signed with self-signed certs. 

Can anyone recommend a provider for codesigning certs? We've been talking to a few companies, but it's like the wild west with huge differences in costs. 

More generally is it considered best practice to sign with a proper key before going to production?

**Comment by AnnaShipman1986** [comment]
@johnkattenhorn Code signing certificates from CA are usually priced higher.

Obtaining the same CA-based code signing certificate from authorized vendors available in the market is an economical option.
 
 SignMyCode offers trusted CA-based various [types of code signing certificates](https://signmycode.com/types-of-codesigning) at affordable prices starting at just $199.99/year.

**Comment by Dvij30** [comment]
Hey @johnkattenhorn yes I think it is considered a very good practice to sign your software with a proper code signing certificate before releasing it to production it provides multiple benefits according to me such as security, trust, legal consideration and lot more. Here are some of the reputed companies for this certificate DiGiCert, GlobalSign, and Thawte all these has different pricing and supports on different platforms.

**Comment by jeromelaban** [comment]
> Can anyone recommend a provider for codesigning certs? We've been talking to a few companies, but it's like the wild west with huge differences in costs.

At Uno Platform, we're using http://www.ssl.com for our code signing certificate (for the nuget packages), and so far it's been reliable.
